.github/workflows/deploy-gcp-tests.yml

```diff
@@ -196,26 +196,43 @@ jobs:
           service_account: 'github-service-account@zealous-zebra.iam.gserviceaccount.com'
           token_format: 'access_token'
 
-      # Before executing any further steps, validate the local state and remote version are the same,
-      # or at least that the local state version is greater than the available cached state version from main.
+      # Find a cached state disk for this job, matching all of:
+      # - disk kind (disk_prefix)
+      # - disk target height (disk_suffix)
+      # - state version (from the source code)
+      #
+      # If there are multiple disks:
+      # - prefer images generated from the `main` branch, then any other branch
+      # - prefer newer images to older images
       #
-      # Aftwards, get the disk name to be used on further steps
-      - name: Validate local state version with cached state
+      # Passes the disk name to subsequent steps using an environmental variable.
+      - name: Find cached state disk
         id: get-disk-name
         run: |
           LOCAL_STATE_VERSION=$(grep -oE "DATABASE_FORMAT_VERSION: .* [0-9]+" "$GITHUB_WORKSPACE/zebra-state/src/constants.rs" | grep -oE "[0-9]+" | tail -n1)
           echo "LOCAL_STATE_VERSION: $LOCAL_STATE_VERSION"
 
-          ZEBRA_STATE_DISK=$(gcloud compute images list --filter="name~${{ inputs.disk_prefix }} AND name~-${{ inputs.disk_suffix }}" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
-          echo "Disk: $ZEBRA_STATE_DISK"
-          echo "Description: $(gcloud compute images describe $ZEBRA_STATE_DISK --format='value(DESCRIPTION)')"
+          # Try to find an image generated from the main branch
+          # Fields are listed in the "Create image from state disk" step
+          CACHED_DISK_NAME=$(gcloud compute images list --filter="name~${{ inputs.disk_prefix }}-main-[0-9a-f]+-v${LOCAL_STATE_VERSION}-${NETWORK}-${{ inputs.disk_suffix }}" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
```

Comment:
```suggestion
          CACHED_DISK_NAME=$(gcloud compute images list --filter="name~${{ inputs.disk_prefix }}-main-[0-9a-f]+-v${LOCAL_STATE_VERSION}-${{ inputs.network }}-${{ inputs.disk_suffix }}" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
```

Reply:
This suggestion won't work because the input is in titlecase, but the disk name is in lowercase.

Instead, I made the network optional in commit f1f6650.